### PR TITLE
autologin: use a helper method to reduce boilerplate

### PR DESCRIPTION
**Pull request by keszybz**
I kept the generated option strings exactly the same. In some cases we put the baud before the device name or placeholder, and in other cases in the opposite order, but maybe there's some reason for this.

The unit file is changed to use just 'aggety' instead of the full path, and the leading whitespace is dropped. Both things should have no effect, but make the resulting file nicer to read.

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -239,67 +239,39 @@ def configure_os_release(state: MkosiState) -> None:
...
+    if not state.config.autologin:
+        return
+
+    with complete_step("Setting up autologin…"):
+        configure_autologin_service(state, "console-getty.service",
+                                    "--noclear --keep-baud console 115200,38400,9600")
+        configure_autologin_service(state, "getty@tty1.service",
+                                    "--noclear -")
+        configure_autologin_service(state, "serial-getty@ttyS0.service",
+                                    "--keep-baud 115200,57600,38400,9600 -")
```
Comment (behrmann) [inline mkosi/__init__.py:270]: Even the longest line is withing the 119 char limit:
```suggestion
        configure_autologin_service(state, "console-getty.service", "--noclear --keep-baud console 115200,38400,9600")
        configure_autologin_service(state, "getty@tty1.service", "--noclear -")
        configure_autologin_service(state, "serial-getty@ttyS0.service", "--keep-baud 115200,57600,38400,9600 -")
```

File: mkosi/__init__.py [hunk 2]
```diff
@@ -239,67 +239,39 @@ def configure_os_release(state: MkosiState) -> None:
...
-                    """\
-                    [Service]
-                    ExecStart=
-                    ExecStart=-/sbin/agetty -o '-f -p -- \\\\u' --autologin root --keep-baud 115200,57600,38400,9600 - $TERM
-                    StandardInput=tty
-                    StandardOutput=tty
-                    """  # noqa: E501
-                )
+            configure_autologin_service(state, "serial-getty@ttyAMA0.service",
+                                        "--keep-baud 115200,57600,38400,9600 -")
```
Comment (behrmann) [inline mkosi/__init__.py:274]: same

File: mkosi/__init__.py [hunk 3]
```diff
@@ -239,67 +239,39 @@ def configure_os_release(state: MkosiState) -> None:
...
+        dropin.write_text(
+            textwrap.dedent(
+                f"""\
                 [Service]
                 ExecStart=
-                ExecStart=-/sbin/agetty -o '-f -p -- \\\\u' --autologin root --noclear --keep-baud console 115200,38400,9600 $TERM
+                ExecStart=-agetty -o '-f -p -- \\\\u' --autologin root {extra} $TERM
                 StandardInput=tty
                 StandardOutput=tty
                 """  # noqa: E501
```
Comment (behrmann) [inline mkosi/__init__.py:255]: That `noqa` shouldn't be necessary any longer, since the `EcecStart` is shorter now.

**Review by behrmann — APPROVED** [review]
LGTM, a few nits.

File: mkosi/__init__.py (hunk 3 above)
Comment (keszybz) [reply mkosi/__init__.py]: I dropped `noqa`.

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**